Allow the configuration file to be specified via an environment variable.

diff --git a/src/gclue-config.c b/src/gclue-config.c
index 7ab2a67..e41f3df 100644
--- a/src/gclue-config.c
+++ b/src/gclue-config.c
@@ -332,6 +332,11 @@ static void
 gclue_config_init (GClueConfig *config)
 {
         GError *error = NULL;
+	const char *config_file_path;
+
+	config_file_path = g_getenv ("GEOCLUE_CONFIG_FILE");
+	if (!config_file_path)
+            config_file_path = CONFIG_FILE_PATH;
 
         config->priv = gclue_config_get_instance_private(config);
         config->priv->key_file = g_key_file_new ();
@@ -341,7 +346,7 @@ gclue_config_init (GClueConfig *config)
                                    &error);
         if (error != NULL) {
                 g_critical ("Failed to load configuration file '%s': %s",
-                            CONFIG_FILE_PATH, error->message);
+                            config_file_path, error->message);
                 g_error_free (error);
 
                 return;
